syntax = "proto3";

option objc_class_prefix = "HLW";

package helloworld;

// The Bitcoin Transcation service definition.
service BitcoinTracker {
  // Adds address tracking for a user.
  rpc AddAddress (AddAddressRequest) returns (AddAddressResponse) {}
  // Removes address tracking for a user.
  rpc RemoveAddress (RemoveAddressRequest) returns (RemoveAddressResponse) {}
  // Detects transactions for a given user between a given set of timestamps.
  rpc DetectTransactions (DetectTransactionsRequest) returns (DetectTransactionsResponse) {} 
}

// The message takes in addresses to be added.
message AddAddressRequest {
  string username = 1;
  repeated string address = 2;
}

// The response contains the list of all addresses for the user.
message AddAddressResponse {
  repeated string address = 1;
}

message DetectTransactionsRequest {
  string username = 1;
  int64 start_timestamp = 2;
  int64 end_timestamp = 3;
}

message Transactions {
   string from_id = 1;
   string to_id = 2;
}

message DetectTransactionsResponse {
  repeated Transactions transactions = 1; 
}
